empiric-package/empiric/core/base_client.py

```diff
@@ -18,44 +18,44 @@ async def _get_nonce(self) -> int:
 
 
 class EmpiricBaseClient(ABC):
+    account_client: AccountClient
+    client: GatewayClient
+    config: IConfig
+    account_contract_address: ADDRESS
+    oracle_controller_address: ADDRESS
+    account_private_key: int
+    signer: StarkCurveSigner
+    oracle_controller_contract: Optional[ADDRESS]
+    account_contract: Optional[ADDRESS]
+
     def __init__(
         self,
-        account_private_key,
-        account_contract_address,
-        network=None,
-        oracle_controller_address=None,
+        account_private_key: ADDRESS,
+        account_contract_address: ADDRESS,
+        network: Network = TESTNET,
+        oracle_controller_address: Optional[ADDRESS] = None,
     ):
-        if network is None:
-            network = NETWORK
-        if oracle_controller_address is None:
-            oracle_controller_address = ORACLE_CONTROLLER_ADDRESS
-
-        if network == TESTNET:
-            chain_id = StarknetChainId.TESTNET
-        elif network == MAINNET:
-            chain_id = StarknetChainId.MAINNET
-        else:
-            raise NotImplementedError(
-                "Empiric.BaseClient: Network not recognized, unknown Chain ID"
-            )
+        raw_config = get_config(network)
+        self.config = raw_config()
```

Comment:
I'm not in love with this because I don't think it conveys that we're initializing an instance, it might make sense to change this to always return a new instance and make this a class method

Reply:
Do you want to change this or leave as is for now @rlkelly ?

Reply:
it's fine for now, we'll clean this up in the next PR